### PR TITLE
coverity: Fix 'unchecked return value' issues.

### PR DESCRIPTION
**Pull request by paul-guo-**

### CODE REVIEW

File: src/backend/cdb/dispatcher/cdbdisp_async.c [hunk 1]
```diff
@@ -818,7 +818,8 @@ checkSegmentAlive(CdbDispatchCmdAsync *pParms)
 static inline void
 send_sequence_response(PGconn *conn, Oid oid, int64 last, int64 cached, int64 increment, bool overflow, bool error)
 {
-	pqPutMsgStart(SEQ_NEXTVAL_QUERY_RESPONSE, false, conn);
+	if (pqPutMsgStart(SEQ_NEXTVAL_QUERY_RESPONSE, false, conn) < 0)
+		elog(ERROR, "Fail to send sequence reponse: %s", PQerrorMessage(conn));
```
Comment (hlinnaka) [inline src/backend/cdb/dispatcher/cdbdisp_async.c:822]: Should this be COMMERROR? Is it OK to throw an error from this function?

File: src/backend/cdb/motion/ic_udpifc.c [hunk 2]
```diff
@@ -1526,7 +1526,7 @@ CleanupMotionUDPIFC(void)
 	uint32		expected = 0;
 
 	/* Shutdown rx thread. */
-	pg_atomic_compare_exchange_u32((pg_atomic_uint32 *) &ic_control_info.shutdown, &expected, 1);
+	(void) pg_atomic_compare_exchange_u32((pg_atomic_uint32 *) &ic_control_info.shutdown, &expected, 1);
```
Comment (hlinnaka) [inline src/backend/cdb/motion/ic_udpifc.c:1529]: How about using pg_atomic_write_u32() instead? Looks like all the references to ic_control_info.shutdown could be replaced with pg_atomic_write_u32() and pg_atomic_read_u32()

File: src/interfaces/libpq/fe-connect.c [hunk 3]
```diff
@@ -3208,8 +3208,8 @@ closePGconn(PGconn *conn)
 		 * Try to send "close connection" message to backend. Ignore any
 		 * error.
 		 */
-		pqPutMsgStart('X', false, conn);
-		pqPutMsgEnd(conn);
+		(void) pqPutMsgStart('X', false, conn);
```
Comment (hlinnaka) [inline src/interfaces/libpq/fe-connect.c:3211]: I'd rather not deviate from upstream on this.

File: src/test/regress/pg_regress.c [hunk 4]
```diff
@@ -318,9 +318,9 @@ static void
 remove_temp(void)
 {
 	Assert(temp_sockdir);
-	unlink(sockself);
-	unlink(socklock);
-	rmdir(temp_sockdir);
+	(void) unlink(sockself);
```
Comment (hlinnaka) [inline src/test/regress/pg_regress.c:321]: Upstream code, I'd prefer to keep it unmodified.

File: src/backend/cdb/dispatcher/cdbdisp_async.c (hunk 1 above)
Comment (paul-guo-) [reply src/backend/cdb/dispatcher/cdbdisp_async.c]: Better ERROR, and it lets QD exit ASAP? I tested this code path by hacking with gdb and got expected error message.

File: src/backend/cdb/motion/ic_udpifc.c (hunk 2 above)
Comment (paul-guo-) [reply src/backend/cdb/motion/ic_udpifc.c]: Checked the code logic. Seems yes. Also change the type to pg_atomic_uint32. Let's see the pipeline output.

File: src/interfaces/libpq/fe-connect.c (hunk 3 above)
Comment (paul-guo-) [reply src/interfaces/libpq/fe-connect.c]: OK.

File: src/test/regress/pg_regress.c (hunk 4 above)
Comment (paul-guo-) [reply src/test/regress/pg_regress.c]: OK.

File: src/backend/commands/analyze.c [hunk 5]
```diff
@@ -4130,8 +4130,8 @@ merge_leaf_stats(VacAttrStatsP stats,
 			totalhll_count++;
 		}
 
-		get_attstatsslot(&hllSlot, heaptupleStats[i], STATISTIC_KIND_HLL,
-						 InvalidOid, ATTSTATSSLOT_VALUES);
+		(void) get_attstatsslot(&hllSlot, heaptupleStats[i], STATISTIC_KIND_HLL,
```
[the remark this answers is not in the document]
Comment (paul-guo-) [reply src/backend/commands/analyze.c:4133]: Yes. As the comment says,`If no matching slot is found, FALSE is returned, and *sslot is zeroed.`, so in that case, hllSlot.nvalues is 0.

**Pull request merged**